mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -44,19 +45,28 @@ class TensorOrMemRefRankOf<list<Type> allowedTypes, list<int> ranks> :
              /*summary=*/"",
              "::mlir::ShapedType">;
 
+class I32ArrayLength<int n> : ConfinedAttr<I32ArrayAttr, [ArrayCount<n>]>;
```

Comment:
Question: why not `DenseI32ArrayAttr`?

Reply:
I didn't have a reason, but now I found out that `ConfinedAttr` only works with  a subclass of `ArrayAttr` . So I cannot confine a dense array

Reply:
Huh, now we know. Cool!